wasm/js/skia_renderer.js

```diff
@@ -4,47 +4,54 @@ Module.onRuntimeInitialized = function () {
         this._ctx = canvas.getContext('2d');
 
         this._realRenderer = _offscreenGL;
-        this['clear'] = function () {
-            const canvas = this._canvas;
-            const glCanvas = _offscreenGL._canvas;
 
-            if (glCanvas.width < canvas.width || glCanvas.height < canvas.height) {
-                glCanvas.width = canvas.width;
-                glCanvas.height = canvas.height;
+        this._drawList = [];
+        this._saveCount = 0;
+
+        this['clear'] = function () {
+            this._drawList = [];  // TODO: what happens on clear if there is a clip???
+            _pendingRenderers.delete(this);
+            if (this._saveCount != 0) {
+                throw "Dunno how to handle a save count in clear()";
```

Comment:
This was more of a question.. Is this scenario possible?

Reply:
Current expectation is clear should always reset/clear the entire frame buffer. Separate from clip, but I think we had one case where we were calling clear after a transform but we fixed that, we should be mindful of not doing that...